src/wg/image/transform.d

```diff
@@ -49,6 +50,49 @@ Image stripMetadata(Image)(ref Image image) if (isImageBuffer!Image)
 // TODO: flip (not in-place, buffer)
 // TODO: rotation (requires destination image buffer, with matching format)
 
+
+///
+void copy(SrcImg, DestImg)(auto ref SrcImg src, ref DestImg dest) nothrow @nogc if (isImage!SrcImg)// TODO: if (something about dest...)
+{
+    // TODO: assert dest is a writable image
+
+    // strongly type the dest buffer if it's soft-typed
+    static if (is(DestImg == ImageBuffer))
+        auto dst = Image!(ElementType!Src)(dest);
+    else
+        alias dst = dest;
+
+    assert(src.width == dst.width && src.height == dst.height);
+
+    enum srcByRow = __traits(compiles, src.row(0));
+    enum destByRow = __traits(compiles, dst.row(0));
+
+    foreach (y; 0 .. src.height)
+    {
+        static if (srcByRow && destByRow)
+        {
+            dst.row(y)[] = src.row(y)[];
+        }
+        else static if (srcByRow)
+        {
+            auto srcRow = src.row(y);
+            foreach (x; 0 .. src.width)
+                dst.at(x, y) = srcRow[x];
+        }
+        else  static if (destByRow)
+        {
```

Comment:
And in this scope `dst` needed `row` only
```
static assert(__traits(compiles, auto destRow = dst.row(y);), "DestImg have no `row(x)` member providing access to row (scanline) of image data.");
```

Reply:
Right, again, template constraint `isImage` guarantees that one or the other is present, and I determine which one above the loop.